src/components/modals/ManageProjectAddresses/AddNewAddress.tsx

```diff
@@ -102,14 +111,19 @@ export const AddNewAddress: FC<IAddNewAddress> = ({
 
 	const validateAddress = async (address: string) => {
 		setLoading(true);
-		if (!isAddress(address)) {
+		if (chainType === ChainType.SOLANA) {
+			if (!isSolanaAddress(address)) {
+				setLoading(false);
+				return 'Invalid Solana address';
+			}
+		} else if (!isAddress(address)) {
 			setLoading(false);
-			return 'Invalid address';
+			return 'Invalid ETH address';
 		}
 		return true;
 	};
 
-	const chainName = getChainName(selectedWallet.networkId);
+	const chainName = getChainName(selectedChain.id, chainType);
```

Comment:
Use general wallet provider (userGeneralWallet) chainName instead.

Reply:
Oh, the name is `useAuthenticationWallet` in your branch still

Reply:
which branch has "userGeneralWallet"? I can't find it

Reply:
It's not merged yet. The `useAuthenticationWallet` is renamed to `userGeneralWallet` in another, but this functionality is not touched. You can go with useAuthenticationWallet so far

Reply:
cc @jainkrati

Reply:
@RamRamez to use useGeneralWallet

Reply:
@aminlatifi it's a wrong idea to use "useGeneralWallet" for fetching chain name here, because useGeneralWallet returns the current chain name that user is on, but in this case, we need the chain name that user has chosen to add wallet address to.
If I use useGeneralWallet, we will have:
![image](https://github.com/Giveth/giveth-dapps-v2/assets/73441857/ea3f7a56-65cb-4d7a-96b7-8a357fcb83e2)

cc @jainkrati